docs/guides/hmac-authentication-in-ruby.md

```diff
@@ -0,0 +1,175 @@
+
+```
+$ dotenv -f .env ruby main.rb
+Got response with status: 401, body: {"message":"Unauthorized"}
+```
+
+## Creating HMAC Signature
+{: .fw-300 }
+
+[The signature](/docs/kb/authentication/hmac#generating-signature) is one of the requirements for forming an API request with HMAC Authentication. The signature is an HMAC256 representation of the request line (a combination of the request method, the request path, and `HTTP/1`.1) and the `Date` header in [RFC 7231](https://www.ietf.org/rfc/rfc7231.txt) format. Carbon will be used to generate the date string for us. The signature must then be converted into a Base64 string so that it can be attached to the `Authorization` header.
+
+The code will look like this: 
+
+```
+<?php // main.php
+
+use Carbon\Carbon;
+
+// ... the rest of the code
+
+datetime = Time.now.httpdate
+request_line = "POST /v2/klikpajak/v1/efaktur/out HTTP/1.1"
+payload = [datetime, request_line].join("\n")
+digest = OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha256'), 'YOUR_MEKARI_API_CLIENT_SECRET', payload)
+signature = Base64.strict_encode64(digest)
+
+puts "hmac username=\"YOUR_MEKARI_API_CLIENT_ID", algorithm=\"hmac-sha256\", headers=\"date request-line\", signature=\"#{signature}\""
+```
+
+If you replace `datetime` with `Wed, 10 Nov 2021 07:24:29 GMT` and run the code, you will get the following result.
+
+```
+$ dotenv -f .env ruby main.rb
+hmac username="YOUR_MEKARI_API_CLIENT_ID", algorithm="hmac-sha256", headers="date request-line", signature="RDHVOBEGBcr86Eyv2Dg42PcwDlTDY4QoJOdkd9w5L0M="
+```
+
+It is important to note that we should not include any credentials in our codebase. This means that we must save the Mekari API client id and client secret that you obtained from the Mekari Developer dashboard to an environment variable. Modern full-stack frameworks, such as Laravel, usually include an `.env` file to make managing environment variables easier. This is also why phpdotenv was installed. We can use this library to move the client id and client secret to the `.env` file. 
+
+```
+# .env file
+MEKARI_API_CLIENT_ID=YOUR_MEKARI_API_CLIENT_ID
+MEKARI_API_CLIENT_SECRET=YOUR_MEKARI_CLIENT_SECRET
+```
+
+Then we use `ENV` to replace the credentials in the code.
+
+```
+// main.rb
+
+require 'time'
+require 'base64'
+require 'openssl'
+require 'faraday'
+
+// ... the rest of the code
+
+datetime = Time.now.httpdate
+request_line = "POST /v2/klikpajak/v1/efaktur/out HTTP/1.1"
+payload = [datetime, request_line].join("\n")
+digest = OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha256'), ENV['MEKARI_API_CLIENT_SECRET'], payload)
+signature = Base64.strict_encode64(digest)
+
+puts "hmac username=\"#{ENV['MEKARI_API_CLIENT_ID']}", algorithm=\"hmac-sha256\", headers=\"date request-line\", signature=\"#{signature}\""
+```
+
+## Wrap It Out
+{: .fw-300 }
+
+It's now time to combine everything we've learned into a single Ruby script.
+
+```
+require 'time'
+require 'base64'
+require 'openssl'
+require 'faraday'
+
```

Comment:
@syafdia We need to add code to load `.env` file in Ruby. Please see example in PHP for the details.